### ISSUE TITLE
Tests fail on TravisCI

### ISSUE THREAD

**Issue by gelisam** [gelisam/hawk#158]
See [here](https://travis-ci.org/gelisam/hawk/jobs/131989554) for example, and the many failures before it. I've managed to reproduce the problem using the following Dockerfile:

```
FROM       ubuntu:latest
MAINTAINER Samuel Gélineau <gelisam@gmail.com>

WORKDIR /root
ENV GHCVER=7.8.2

RUN apt-get update
RUN apt-get install -y software-properties-common
RUN add-apt-repository -y ppa:hvr/ghc
RUN apt-get update
RUN apt-get install -y git
RUN apt-get install -y cabal-install-1.18 happy-1.19.3 ghc-$GHCVER
RUN git clone --depth=50 --branch=1.1.2 https://github.com/gelisam/hawk.git gelisam/hawk
WORKDIR /root/gelisam/hawk
RUN git checkout -qf d22ae9143fef2b2e78e3ee433ec0be6d2c496732
ENV PATH=/opt/happy/1.19.3/bin:/opt/ghc/$GHCVER/bin:$PATH
RUN cabal-1.18 update
RUN cabal-1.18 install --only-dependencies --enable-tests --enable-benchmarks
RUN cabal-1.18 configure --enable-tests --enable-benchmarks -v2
RUN cabal-1.18 build
RUN cabal-1.18 test

CMD ["bash"]
```

**Comment by gelisam** [comment]
I didn't do a very good job following our git workflow this time... Nor [last time](https://github.com/gelisam/hawk/issues/147#issuecomment-118842472) now that I think of it. I rewrote the history (if you cloned this in the last few days, sorry!) to put master and stable back into their pre-1.1.2 state, and I'm now doing this properly on the release-1.1.2 branch.

**Comment by gelisam** [comment]
So I can see in the output that the doctest tests pass and that it's the "Test suite reference" part which fails, but if I comment out the doctests and leave only the test suite reference, it passes. There an interaction between my tests?

**Comment by gelisam** [comment]
I remember that doctest tests can interact with each other because ghci's state is kept between tests, and I've had to reorder some of the tests over the years to avoid mysterious deadlocks. I don't understand how doctest could have an impact on non-doctest tests.

**Comment by gelisam** [comment]
Great, now TravisCI [refuses to run our tests](https://travis-ci.org/gelisam/hawk/builds/148932169), I guess we have to upgrade the format of our `.travis.yml` file or something.

    ERROR: An error occured while trying to parse your .travis.yml file.
    
    Please make sure that the file is valid YAML.
    
    http://lint.travis-ci.org can check your .travis.yml.
    
    The log message was: Build config file had a parse error: "did not find expected key while parsing a     block mapping at line 9 column 1".

**Comment by gelisam** [comment]
"Fixed" in #173 by disabling the tests. I'll re-enable them in #174.
